src/server/game/Battlegrounds/Battleground.cpp

```diff
@@ -667,22 +668,24 @@ void Battleground::SendChatMessage(Creature* source, uint8 textId, WorldObject*
     sCreatureTextMgr->SendChat(source, textId, target);
 }
 
-void Battleground::PlaySoundToAll(uint32 SoundID)
+void Battleground::PlaySoundToAll(uint32 soundID)
 {
-    WorldPacket data;
-    sBattlegroundMgr->BuildPlaySoundPacket(&data, SoundID);
-    SendPacketToAll(&data);
+    WorldPackets::Misc::PlaySound playSound;
+    sBattlegroundMgr->BuildPlaySoundPacket(&playSound, soundID);
+    SendPacketToAll(playSound.Write());
 }
```

Comment:
I'd suggest drop the BattlegroundMgr::BuildPlaySoundPacket and use the constructor of PlaySound that you already added here:

```c++
    WorldPackets::Misc::PlaySound playSound(ObjectGuid::Empty, soundID);
    SendPacketToAll(playSound.Write());
```